### PR TITLE
fix(web-components): remove final dependencies on fast-foundation

### PR DESCRIPTION
**Pull request by chrisdholt**
This is the final PR to fully remove the dependency on the fast-foundation project. There are two utilities being moved here (along with some cleanup found while changing/removing imports): the `display` helper and the `match-media-stylesheet` behavior and corresponding code.

### CODE REVIEW

**Review by eljefe223 — APPROVED** [review]

**Comment by fabricteam** [comment]
## 🕵 fluentui-web-components-v3 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

File: change/@fluentui-web-components-86d45358-8f3e-454f-92b1-96b3ea94e9a0.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "fix(web-components): remove final dependencies on fast-foundation",
+  "packageName": "@fluentui/web-components",
+  "email": "=",
```
Comment (radium-v) [inline change/@fluentui-web-components-86d45358-8f3e-454f-92b1-96b3ea94e9a0.json:5]: Should this be set?

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                   |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>global-context</samp> <br /> <abbr title='bundle-size/createContext.fixture.js'>createContext</abbr>                          |          `533 B`<br />`338 B` |
| <samp>global-context</samp> <br /> <abbr title='bundle-size/createContextSelector.fixture.js'>createContextSelector</abbr>          |          `554 B`<br />`348 B` |
| <samp>priority-overflow</samp> <br /> <abbr title='bundle-size/overflow-manager.fixture.js'>createOverflowManager</abbr>            |    `3.194 kB`<br />`1.315 kB` |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr> |   `85.92 kB`<br />`25.987 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/Badge.fixture.js'>Badge</abbr>                                             |    `23.78 kB`<br />`7.544 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/CounterBadge.fixture.js'>CounterBadge</abbr>                               |    `24.67 kB`<br />`7.829 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/PresenseBadge.fixture.js'>PresenceBadge</abbr>                             |   `25.849 kB`<br />`7.546 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                          |   `36.883 kB`<br />`9.427 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                          |    `44.058 kB`<br />`10.9 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                  |  `41.616 kB`<br />`10.787 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                |  `50.045 kB`<br />`12.351 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                              |  `55.247 kB`<br />`11.342 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                      |  `83.371 kB`<br />`23.931 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                |    `78.14 kB`<br />`22.44 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                    |     `9.04 kB`<br />`3.793 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                    |   `10.979 kB`<br />`4.506 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                  |     `9.847 kB`<br />`4.14 kB` |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/Checkbox.fixture.js'>Checkbox</abbr>                                    |  `33.139 kB`<br />`10.361 kB` |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/CheckboxField.fixture.js'>CheckboxField</abbr>                          |  `40.044 kB`<br />`12.241 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>          |  `89.767 kB`<br />`26.682 kB` |
| <samp>react-divider</samp> <br /> <abbr title='bundle-size/Divider.fixture.js'>Divider</abbr>                                       |   `17.105 kB`<br />`6.197 kB` |
| <samp>react-field</samp> <br /> <abbr title='bundle-size/Field.fixture.js'>Field</abbr>                                             |   `17.721 kB`<br />`6.563 kB` |
| <samp>react-image</samp> <br /> <abbr title='bundle-size/Image.fixture.js'>Image</abbr>                                             |   `11.259 kB`<br />`4.465 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                             |   `24.848 kB`<br />`8.173 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/InputField.fixture.js'>InputField</abbr>                                   |  `35.106 kB`<br />`11.123 kB` |
| <samp>react-label</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Label</abbr>                                             |    `9.817 kB`<br />`4.059 kB` |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr>                                               |   `12.306 kB`<br />`5.083 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                | `126.252 kB`<br />`38.699 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>   | `129.321 kB`<br />`39.216 kB` |
| <samp>react-overflow</samp> <br /> <abbr title='bundle-size/hooks.fixture.js'>hooks only</abbr>                                     |   `11.214 kB`<br />`4.247 kB` |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                       | `113.927 kB`<br />`35.085 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                          |   `11.377 kB`<br />`4.159 kB` |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePositioning</abbr>                          |   `22.857 kB`<br />`8.512 kB` |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/ProgressBar.fixture.js'>ProgressBar</abbr>                              |   `13.573 kB`<br />`5.241 kB` |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/ProgressField.fixture.js'>ProgressField</abbr>                          |   `23.921 kB`<br />`8.394 kB` |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr>                        |   `17.116 kB`<br />`6.334 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                                             |  `32.724 kB`<br />`10.608 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                                   |   `14.885 kB`<br />`5.904 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroupField.fixture.js'>RadioGroupField</abbr>                         |   `25.011 kB`<br />`9.086 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                          |   `24.193 kB`<br />`8.314 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/SelectField.fixture.js'>SelectField</abbr>                                |  `33.432 kB`<br />`10.811 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                          |  `33.086 kB`<br />`10.576 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/SliderField.fixture.js'>SliderField</abbr>                                |   `42.97 kB`<br />`13.503 kB` |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr>                                       |    `20.615 kB`<br />`6.69 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                          |   `28.273 kB`<br />`8.781 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/SwitchField.fixture.js'>SwitchField</abbr>                                |  `35.058 kB`<br />`10.663 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Default.fixture.js'>Text - Default</abbr>                              |   `12.261 kB`<br />`4.824 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Wrappers.fixture.js'>Text - Wrappers</abbr>                            |   `15.571 kB`<br />`5.245 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr>                                    |   `26.331 kB`<br />`8.598 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/TextareaField.fixture.js'>TextareaField</abbr>                          |  `36.585 kB`<br />`11.532 kB` |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/SingleThemeToken.fixture.js'>Single theme token import</abbr>              |            `69 B`<br />`89 B` |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/TeamsAllThemes.fixture.js'>Teams: all themes</abbr>                        |   `30.908 kB`<br />`6.725 kB` |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/TeamsLightTheme.fixture.js'>Teams: Light theme</abbr>                      |   `17.776 kB`<br />`5.144 kB` |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                       |  `45.962 kB`<br />`16.154 kB` |
| <samp>react-utilities</samp> <br /> <abbr title='bundle-size/SSRProvider.fixture.js'>SSRProvider</abbr>                             |          `180 B`<br />`160 B` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/bea6d29b58df832fd14449ad901aa8afd7b3fd7c'>bea6d29b58df832fd14449ad901aa8afd7b3fd7c</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-web-components-86d45358-8f3e-454f-92b1-96b3ea94e9a0.json (hunk 1 above)
Comment (chrisdholt) [reply change/@fluentui-web-components-86d45358-8f3e-454f-92b1-96b3ea94e9a0.json:5]: I don't think it matters - I'll look and see why it's "not" setting.

File: apps/vr-tests-web-components/src/utilities/WCThemeDecorator.tsx [hunk 2]
```diff
@@ -2,12 +2,9 @@ import * as React from 'react';
 import { StoryContext } from '@storybook/addons';
```
Comment (fabricteam) [inline apps/vr-tests-web-components/src/utilities/WCThemeDecorator.tsx:2]: ## [🕵 fluentui-web-components-v3 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31009&commitId=2d97e683606e80c5e33fe1e9d9fa0a6290364423&env=prod&clientType=FLUENTUI)


  <details><summary><b>Text</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Text.Block.normal.chromium.png|995| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

Comment (chrisdholt) [reply apps/vr-tests-web-components/src/utilities/WCThemeDecorator.tsx:2]: This didn't update with the last one - the background *is expected*

**Review by Hotell — APPROVED** [review]
approving v-build related changes

**Pull request merged**